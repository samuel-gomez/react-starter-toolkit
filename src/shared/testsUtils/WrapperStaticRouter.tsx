/* eslint-disable react/display-name */
import { ReactNode } from 'react';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';

const WrapperStaticRouter =
  (route = '/') =>
  ({ children }: { children: ReactNode }) =>
    <MemoryRouterProvider url={route}>{children}</MemoryRouterProvider>;

export default WrapperStaticRouter;
