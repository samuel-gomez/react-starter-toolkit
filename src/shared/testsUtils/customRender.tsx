/* eslint-disable react/display-name */
import { render, RenderOptions } from '@testing-library/react';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';

import FetchProvider from 'providers/FetchProvider';
import UserProvider from 'providers/UserProvider';
import { ReactElement, ReactNode } from 'react';
import { MOCK_API_URL } from '.';

type TMockProvider = {
  [x: string]: Record<string, unknown | number | string> | string;
};

const MockProviders =
  ({ role = '', ...testMock }: TMockProvider) =>
  ({ children }: { children: ReactNode }) => {
    const useOidcAccessTokenFn = jest.fn().mockReturnValue({ accessToken: 'accessToken' });
    const useOidcUserFn = jest.fn().mockReturnValueOnce({ oidcUser: { member_of: [`CN=${role}`], name: 'Samuel Gomez' } });
    return (
      <UserProvider useOidcUserFn={useOidcUserFn}>
        <FetchProvider
          apiUrl={MOCK_API_URL}
          fetchConfig={{
            headers: { testMock: JSON.stringify(testMock) },
          }}
          useOidcAccessTokenFn={useOidcAccessTokenFn}
        >
          <MemoryRouterProvider url="">{children}</MemoryRouterProvider>
        </FetchProvider>
      </UserProvider>
    );
  };

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>, testMock: TMockProvider = { role: '' }) =>
  render(ui, { wrapper: MockProviders(testMock), ...options });

// re-export everything
export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

// override render method
export { customRender as render };
