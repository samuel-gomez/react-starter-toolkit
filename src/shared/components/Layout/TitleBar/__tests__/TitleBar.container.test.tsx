import { emptyFunction, renderWithWrapperStaticRouter } from 'shared/testsUtils';
import TitleBarEnhanced from '../';

const defaultProps = { title: 'title', handleClick: emptyFunction };

const createRender = (customProps = {}) => {
  const actualProps = { ...defaultProps, ...customProps };
  return renderWithWrapperStaticRouter(<TitleBarEnhanced {...actualProps} />);
};

describe('Component <TitleBarEnhanced />', () => {
  it.each`
    customProps                                                                                   | classNameExpected
    ${undefined}                                                                                  | ${'.af-title-bar'}
    ${{ className: 'otherclassname' }}                                                            | ${'.otherclassname'}
    ${{ classModifier: 'red' }}                                                                   | ${'.af-title-bar--red'}
    ${{ backHome: true }}                                                                         | ${'.af-title-bar--backhome'}
    ${{ backHome: true, classModifier: 'custom' }}                                                | ${'.af-title-bar--backhome.af-title-bar--custom'}
    ${{ className: 'otherclassname', backHome: true, classModifier: 'custom' }}                   | ${'.otherclassname--backhome.otherclassname--custom'}
    ${{ className: 'otherclassname', backHome: true, classModifier: 'custom', fullScreen: true }} | ${'.otherclassname--backhome.otherclassname--custom.otherclassname--fullscreen'}
  `('Should render TitleBarEnhanced with customProps: $customProps', ({ customProps, classNameExpected }) => {
    const { asFragment, container } = createRender(customProps);
    expect(container.querySelector(classNameExpected)).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
