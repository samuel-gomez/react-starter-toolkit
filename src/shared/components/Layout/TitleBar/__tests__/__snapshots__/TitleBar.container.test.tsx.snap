// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <TitleBarEnhanced /> Should render TitleBarEnhanced with customProps: {"backHome": true, "classModifier": "custom", "className": "otherclassname", "fullScreen": true} 1`] = `
<DocumentFragment>
  <div
    class="otherclassname otherclassname--custom otherclassname--backhome otherclassname--fullscreen"
  >
    <div
      class="container af-title-bar__wrapper"
    >
      <div
        class="burger-container"
      >
        <a
          aria-controls="mainmenu"
          aria-haspopup="true"
          class="btn af-title-bar__mobile-menu af-btn--circle"
          href="/#"
          id="togglemenu"
          role="button"
          tabindex="0"
          title="Toggle menu"
        >
          <i
            class="glyphicon glyphicon-menu-hamburger"
          />
        </a>
      </div>
      <h1
        class="af-title-bar__title"
      >
        title
      </h1>
      <a
        aria-label="Retour à l'accueil"
        class="btn af-btn--circle"
        href="/"
      >
        <i
          class="glyphicon glyphicon-home"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Component <TitleBarEnhanced /> Should render TitleBarEnhanced with customProps: {"backHome": true, "classModifier": "custom", "className": "otherclassname"} 1`] = `
<DocumentFragment>
  <div
    class="otherclassname otherclassname--custom otherclassname--backhome"
  >
    <div
      class="container af-title-bar__wrapper"
    >
      <div
        class="burger-container"
      >
        <a
          aria-controls="mainmenu"
          aria-haspopup="true"
          class="btn af-title-bar__mobile-menu af-btn--circle"
          href="/#"
          id="togglemenu"
          role="button"
          tabindex="0"
          title="Toggle menu"
        >
          <i
            class="glyphicon glyphicon-menu-hamburger"
          />
        </a>
      </div>
      <h1
        class="af-title-bar__title"
      >
        title
      </h1>
      <a
        aria-label="Retour à l'accueil"
        class="btn af-btn--circle"
        href="/"
      >
        <i
          class="glyphicon glyphicon-home"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Component <TitleBarEnhanced /> Should render TitleBarEnhanced with customProps: {"backHome": true, "classModifier": "custom"} 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar af-title-bar--custom af-title-bar--backhome"
  >
    <div
      class="container af-title-bar__wrapper"
    >
      <div
        class="burger-container"
      >
        <a
          aria-controls="mainmenu"
          aria-haspopup="true"
          class="btn af-title-bar__mobile-menu af-btn--circle"
          href="/#"
          id="togglemenu"
          role="button"
          tabindex="0"
          title="Toggle menu"
        >
          <i
            class="glyphicon glyphicon-menu-hamburger"
          />
        </a>
      </div>
      <h1
        class="af-title-bar__title"
      >
        title
      </h1>
      <a
        aria-label="Retour à l'accueil"
        class="btn af-btn--circle"
        href="/"
      >
        <i
          class="glyphicon glyphicon-home"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Component <TitleBarEnhanced /> Should render TitleBarEnhanced with customProps: {"backHome": true} 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar af-title-bar--backhome"
  >
    <div
      class="container af-title-bar__wrapper"
    >
      <div
        class="burger-container"
      >
        <a
          aria-controls="mainmenu"
          aria-haspopup="true"
          class="btn af-title-bar__mobile-menu af-btn--circle"
          href="/#"
          id="togglemenu"
          role="button"
          tabindex="0"
          title="Toggle menu"
        >
          <i
            class="glyphicon glyphicon-menu-hamburger"
          />
        </a>
      </div>
      <h1
        class="af-title-bar__title"
      >
        title
      </h1>
      <a
        aria-label="Retour à l'accueil"
        class="btn af-btn--circle"
        href="/"
      >
        <i
          class="glyphicon glyphicon-home"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Component <TitleBarEnhanced /> Should render TitleBarEnhanced with customProps: {"classModifier": "red"} 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar af-title-bar--red"
  >
    <div
      class="container af-title-bar__wrapper"
    >
      <div
        class="burger-container"
      >
        <a
          aria-controls="mainmenu"
          aria-haspopup="true"
          class="btn af-title-bar__mobile-menu af-btn--circle"
          href="/#"
          id="togglemenu"
          role="button"
          tabindex="0"
          title="Toggle menu"
        >
          <i
            class="glyphicon glyphicon-menu-hamburger"
          />
        </a>
      </div>
      <h1
        class="af-title-bar__title"
      >
        title
      </h1>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Component <TitleBarEnhanced /> Should render TitleBarEnhanced with customProps: {"className": "otherclassname"} 1`] = `
<DocumentFragment>
  <div
    class="otherclassname"
  >
    <div
      class="container af-title-bar__wrapper"
    >
      <div
        class="burger-container"
      >
        <a
          aria-controls="mainmenu"
          aria-haspopup="true"
          class="btn af-title-bar__mobile-menu af-btn--circle"
          href="/#"
          id="togglemenu"
          role="button"
          tabindex="0"
          title="Toggle menu"
        >
          <i
            class="glyphicon glyphicon-menu-hamburger"
          />
        </a>
      </div>
      <h1
        class="af-title-bar__title"
      >
        title
      </h1>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Component <TitleBarEnhanced /> Should render TitleBarEnhanced with customProps: undefined 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar"
  >
    <div
      class="container af-title-bar__wrapper"
    >
      <div
        class="burger-container"
      >
        <a
          aria-controls="mainmenu"
          aria-haspopup="true"
          class="btn af-title-bar__mobile-menu af-btn--circle"
          href="/#"
          id="togglemenu"
          role="button"
          tabindex="0"
          title="Toggle menu"
        >
          <i
            class="glyphicon glyphicon-menu-hamburger"
          />
        </a>
      </div>
      <h1
        class="af-title-bar__title"
      >
        title
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
